tests/performance-test/performance-test.sh

```diff
@@ -49,91 +49,170 @@ ellipse(){
-
-    if ! oc project sa-telemetry; then 
+# basic check if necessary resources exist. DOES NOT verify they work correctly
+check_resources(){
+    printf "\n*** Performing Resource Checks ***\n"
+    if ! oc get project sa-telemetry; then 
         echo "Error: SAF does not exist on cluster. Deploy SAF before running performance test" 1>&2
         exit 1
     fi
+
+    if ! oc get project openshift-monitoring; then 
+        echo "Error: openshift-monitoring not enabled on host. Continuing without it will result \
+        in reduced functionality of performance test. Would you like to continue without \
+        it? [y/n]"
+        read RESP
+        case $RESP in
+            "y") 
+            ;;
+            *)
+                exit 1
+            ;;
+        esac
+    fi
+
     if ! GRAF_HOST=$(oc get routes --field-selector metadata.name=grafana -o jsonpath="{.items[0].spec.host}") 2> /dev/null; then 
         echo "Error: cannot find Grafana instance in cluster. Has it been deployed?" 1>&2
         exit 1
     fi
 
-    SAF_PROM_HOST=$(oc get routes --field-selector metadata.name=prometheus -o jsonpath="{.items[0].spec.host}")
+    local datasources=$(curl --silent -X GET "http://${GRAF_HOST}/api/datasources")
+    if ! echo $datasources | grep -q "OCPPrometheus"; then
+        echo "Error: unable to find Grafana datasource OCPPrometheus"
+        exit 1
+    fi
 
-    SAF_DATASOURCE=$(cat << EOF
-        {
-            "name"  :   "SAFPrometheus",
-            "type"  :   "prometheus",
-            "url"   :   "http://$SAF_PROM_HOST",
-            "access":   "direct",
-            "basicAuth" :false,
-            "jsonData": {
-                "timeInterval"  :   "1s"
-            }
-        }
-EOF
-)
+    if ! echo $datasources | grep -q "SAFPrometheus"; then
+        echo "Error: unable to find Grafana datasource SAFPrometheus"
+        exit 1
+    fi
+
+    if ! echo $datasources | grep -q "SAFElasticsearch"; then
+        echo "Error: unable to find Grafana datasource SAFElasticsearch"
+        exit 1
+    fi
 }
 
+# Post dashboards to grafana - overwrites if they already exist
 post_dashboards(){
-    echo "Creating new dashboards in Grafana"
+    printf "\n*** Creating new dashboards in Grafana ***\n"
     curl --silent --output /dev/null -d "{\"overwrite\": true, \"dashboard\": $(cat ./grafana/perftest-dashboard.json)}" \
         -H 'Content-Type: application/json' "$GRAF_HOST/api/dashboards/db"
         
     curl --silent --output /dev/null -d "{\"overwrite\": true, \"dashboard\": $(cat ./grafana/prom2-dashboard.json)}" \
         -H 'Content-Type: application/json' "$GRAF_HOST/api/dashboards/db"
 }
 
-# Execute
+# Delete recorded events in Elastic Search
+delete_es_events(){
+    local esPod=$(oc get elasticsearch elasticsearch -o jsonpath='{.status.pods.master.ready[0]}')
+    local secretPath="/etc/elasticsearch/secret/"
+    local esQuery=$(cat <<EOF
```

Comment:
Just a personal preference, but I'd put all this in one wide line, or at LEAST collapse all the `}}}}}` into one line at the end. I bet others might disagree :)

Reply:
I'd disagree :)

Reply:
That's an interesting way to do it